create_jenkins_job.py

```diff
@@ -114,73 +116,89 @@ def main(argv=None):
         jenkins_kwargs['dry_run'] = True
 
-    # configure os specific jobs
-    for os_name in sorted(os_configs.keys()):
+    def create_job(os_name, job_name, template_file, additional_dict):
         job_data = dict(data)
         job_data['os_name'] = os_name
         job_data.update(os_configs[os_name])
+        job_data.update(additional_dict)
+        job_config = expand_template(template_file, job_data)
+        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
 
+    # configure os specific jobs
+    for os_name in sorted(os_configs.keys()):
         # configure manual triggered job
-        job_name = 'ci_' + os_name
-        job_data['cmake_build_type'] = 'None'
-        job_config = expand_template('ci_job.xml.em', job_data)
-        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
+        create_job(os_name, 'ci_' + os_name, 'ci_job.xml.em', {
+            'cmake_build_type': 'None',
+        })
 
         # configure a manual version of the packaging job
-        job_name = 'ci_packaging_' + os_name
-        job_data['cmake_build_type'] = 'RelWithDebInfo'
-        job_data['test_bridge_default'] = 'true'
-        job_config = expand_template('packaging_job.xml.em', job_data)
-        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
-
-        # all following jobs are triggered nightly with email notification
-        job_data['time_trigger_spec'] = '30 7 * * *'
-        # for now, skip emailing about Windows failures
-        job_data['mailer_recipients'] = 'ros2-buildfarm@googlegroups.com'
+        create_job(os_name, 'ci_packaging_' + os_name, 'packaging_job.xml.em', {
+            'cmake_build_type': 'RelWithDebInfo',
+            'test_bridge_default': 'true',
+        })
 
         # configure packaging job
-        job_name = 'packaging_' + os_name
-        job_data['cmake_build_type'] = 'RelWithDebInfo'
-        job_data['test_bridge_default'] = 'true'
-        job_config = expand_template('packaging_job.xml.em', job_data)
-        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
+        create_job(os_name, 'packaging_' + os_name, 'packaging_job.xml.em', {
+            'cmake_build_type': 'RelWithDebInfo',
+            'test_bridge_default': 'true',
+            'time_trigger_spec': PERIODIC_JOB_SPEC,
+            'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
+        })
 
-        # keeping the paths on Windows shorter
-        os_name = os_name.replace('windows', 'win')
         # configure nightly triggered job
-        job_name = 'nightly_' + os_name + '_debug'
-        if os_name == 'win':
-            job_name = job_name[0:-2]
-        job_data['cmake_build_type'] = 'Debug'
-        job_config = expand_template('ci_job.xml.em', job_data)
-        configure_job(jenkins, job_name, job_config, **jenkins_kwargs)
+        short_name = 'nightly_' + os_name + '_debug'
+        if os_name == 'windows':
+            short_name = 'nightly_win_deb'
```

Comment:
the previous logic was creating the job_name the same way for everyone and then modifying it to make it shorter for windows. This new approach replace the variable content with a hard-coded string literal discarding the previous content so I think it's more likely that the naming becomes inconsistent if we change one but not the other